eslint (2.0) erroneously complains about correct spacing

Test files (renamed to have ending .txt for github to accept them as issue attachment):

[test.js.txt](https://github.com/eslint/eslint/files/129084/test.js.txt)
[.eslintrc.js.txt](https://github.com/eslint/eslint/files/129085/default.eslintrc.js.txt)

The code 

```
const runImapServer = function (imapAccountSettingsHash) {
    return Promise.resolve()
    .then(() => {
        [
            'APPEND', 'CREATE', 'RENAME', 'COPY', 'UID COPY', 'EXPUNGE', 'STORE', 'DELETE'
        ].forEach(command => {
            // ...
        });
    });
};
```

causes warnings/errors of the type

    Expected indentation of 12 space characters but found 8   indent

for the lines containing the (strings) array. **However**, not only is the spacing correct, what's even more telling that this is a bug is that if I change the function expression from being assigned to a `const` to instead assigning it to `module.exports.someVar` the complaining stops and the spaces are accepted! See the attached file for the full example.

This is what I get when I run `eslint test.js`:
```

$ eslint test.js

C:\Users\Michael\OneDrive\Projects\One\test.js
  6:9   warning  Expected indentation of 12 space characters but found 8   indent
  7:13  warning  Expected indentation of 16 space characters but found 12  indent
  8:9   warning  Expected indentation of 12 space characters but found 8   indent

✖ 3 problems (0 errors, 3 warnings)
```

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -429,13 +429,11 @@ module.exports = function(context) {
             nodeIndent = getNodeIndent(effectiveParent);
             if (parentVarNode && parentVarNode.loc.start.line !== node.loc.start.line) {
                 if (parent.type !== "VariableDeclarator" || parentVarNode === parentVarNode.parent.declarations[0]) {
-                    if (options.VariableDeclarator[parentVarNode.parent.kind] === 0 && (parent.type === "ObjectExpression" || parent.type === "ArrayExpression")) {
-                        nodeIndent = nodeIndent + indentSize;
-                    } else {
+                    if (parentVarNode.loc.start.line === effectiveParent.loc.start.line) {
                         nodeIndent = nodeIndent + (indentSize * options.VariableDeclarator[parentVarNode.parent.kind]);
+                    } else if (parent.type === "ObjectExpression" || parent.type === "ArrayExpression") {
+                        nodeIndent = nodeIndent + indentSize;
                     }
-                } else if (parent.loc.start.line !== node.loc.start.line && parentVarNode === parentVarNode.parent.declarations[0]) {
-                    nodeIndent = nodeIndent + indentSize;
                 }
             } else if (!parentVarNode && !isFirstArrayElementOnSameLine(parent) && effectiveParent.type !== "MemberExpression" && effectiveParent.type !== "ExpressionStatement" && effectiveParent.type !== "AssignmentExpression" && effectiveParent.type !== "Property") {
                 nodeIndent = nodeIndent + indentSize;

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -50,7 +50,6 @@ function expectedErrors(indentType, errors) {
 
 var ruleTester = new RuleTester();
 ruleTester.run("indent", rule, {
-
     valid: [
         {
             code:
@@ -1042,6 +1041,19 @@ ruleTester.run("indent", rule, {
             "    }\n" +
             "];\n",
             options: [4, {"VariableDeclarator": 0, "SwitchCase": 1}]
+        },
+        {
+            code:
+            "const func = function (opts) {\n" +
+            "    return Promise.resolve()\n" +
+            "    .then(() => {\n" +
+            "        [\n" +
+            "            'ONE', 'TWO'\n" +
+            "        ].forEach(command => { doSomething(); });\n" +
+            "    });\n" +
+            "};",
+            parserOptions: { ecmaVersion: 6 },
+            options: [4, {"SwitchCase": 1}]
         }
     ],
     invalid: [

